movies/4.create-movie.md

```diff
@@ -0,0 +1,212 @@
+# Create a movie
+
+## Story
+
+As an API client I want to create a new movie resource
+
+{% hint style='info' %}
+Several possibilities exist, to define operations with Hydra. In this example we will describe a
+custom type `mov:Movies` in the API documenation and add a supported operation there.
+{% endhint %}
+
+## API behaviour
+
+```http
+GET https://hydra-movies.herokuapp.com/
+```
+
+Response body (excerpt):
+
+```json
+{
+  "@id": "/",
+  "collection": {
+    "@id": "movies",
+    "@type": [
+      "Collection",
+      "mov:Movies"
+    ]
+  }
+}
+```
+
+Api documentation (excerpt):
+
+```json
+{
+  "@id": "/doc",
+  "supportedClass": [
+    {
+      "@id": "mov:Movies",
+      "@type": "Class",
+      "supportedOperation": {
+        "@id": "mov:create-movie",
+        "@type": [
+          "Operation",
+          "schema:CreateAction"
+        ],
+        "title": "Create movie",
+        "description": "Creates a new movie",
+        "method": "POST",
+        "expects": "schema:Movie",
+        "returns": "schema:Movie"
+      }
+    },
+    {
+      "@id": "schema:Movie",
+      "@type": "Class",
+      "title": "Movie",
+      "description": "A single movie",
+      "supportedProperty": [
+        {
+          "property": "schema:name",
+          "title": "name",
+          "description": "The movie's name"
+        },
+        {
+          "property": "schema:description",
+          "title": "description",
+          "description": "Textual description of the movie."
+        },
+        {
+          "property": "schema:duration",
+          "title": "duration",
+          "description": "The movie's duration in ISO 8601 date format."
+        }
+      ]
+    }
+  ]
+}
+```
+
+```http
+POST https://hydra-movies.herokuapp.com/movies
+Content-Type: application/ld+json
+
+{
+  "@context": "https://schema.org/",
+  "@type": "Movie",
+  "name": "Pulp Fiction",
+  "description": "Pulp Fiction is a 1994 American crime film. It tells several stories of criminal Los Angeles.",
+  "duration": "PT2H34M"
+}
+```
+
+Response (excerpt):
+
+```http
+HTTP/1.1 201 Created
+Location: /movies/d517cae6-6cdc-11e9-a923-1681be663d3e
+Content-Type: application/ld+json
+
+{
+  "@context": "https://hydra-movies.herokuapp.com/context.jsonld",
+  "@type": "schema:Movie",
+  "@id": "/movies/d517cae6-6cdc-11e9-a923-1681be663d3e",
+  "schema:name": "Pulp Fiction"
+}
+```
+
+
+### Explanation
+
+```
+"collection": {
+    "@id": "movies",
+    "@type": [
+      "Collection",
+      "mov:Movies"
+    ]
+  }
+```
+
+By loading the API entry point a client came across a resource of type `mov:Movies`. To understand what it can do with it, it consults the API documentation.
```

Comment:
```suggestion
By loading the API entry point a client came across a resource of type `mov:Movies`. To understand what it can do with it, it may consults the API documentation in case there are no in-band operations provided.
```

Reply:
Proposal from call: Due to the fact that there are no inline hypermedia controls, the client looks to the API documentation ...

Reply:
rephrased